packages/core/src/lib/chains/algorand/AlgorandConnect.ts

```diff
@@ -0,0 +1,143 @@
+import {BridgeNetworks} from "src/lib/common/networks/networks";
+import BigNumber from "bignumber.js";
+import {bridgeDeposit, bridgeUSDC} from "./transactions";
+import SendRawTransaction from "algosdk/dist/types/client/v2/algod/sendRawTransaction";
+import {RoutingDefault} from "src/lib/common";
+import {AlgorandAccountsStore} from "./AccountsStore";
+
+export class AlgorandConnect {
+    public readonly clientIndexer: algosdk.Indexer;
+    public readonly assetsRepo: AssetsRepository;
+    public readonly config: AlgorandConfig;
+    public readonly client: algosdk.Algodv2;
+    public readonly accountsStore: AlgorandAccountsStore;
+    _lastTxnHash = "";
+
+    constructor(config: AlgorandConfig) {
+        this.config = config;
+        this.client = this.getAlgodClient();
+        this.clientIndexer = this.getAlgodIndexer();
+        this.assetsRepo = new AssetsRepository(this.client);
+        this.accountsStore = new AlgorandAccountsStore(this.client);
+
+        config.assets.map(
+            (conf: AlgorandStandardAssetConfig | AlgorandAssetConfig) => {
+                if ((conf as AlgorandStandardAssetConfig).assetId) {
+                    return this.assetsRepo.addStandardAsset(
+                        (conf as AlgorandStandardAssetConfig).assetId,
+                        conf
+                    );
+                }
+            }
+        );
+    }
+
+    /**
+   *
+   * @param sourceAddress
+   * @param destinationNetwork
+   * @param destinationAdress
+   * @param tokenSymbol
+   * @param amount
+   * @returns
+   */
+    public async bridgeTransactions(
+        sourceAddress: string,
+        destinationNetwork: BridgeNetworks,
+        destinationAdress: string,
+        tokenSymbol: string,
+        amount: bigint
+    ): Promise<algosdk.Transaction[]> {
+        const token = this.assetsRepo.getAsset(tokenSymbol);
+        if (!token) return Promise.reject("Token unsupported");
+        if (!token.destinationSymbol[destinationNetwork])
+            return Promise.reject("Token unsupported");
+
+        const depositAddress = this.config.bridgeAccounts.usdcDeposit;
+        const routing = RoutingDefault();
+        routing.from.address = sourceAddress;
+        routing.from.token = tokenSymbol;
+        routing.from.network = BridgeNetworks.algorand.toString().toLowerCase();
+        routing.to.address = destinationAdress;
+        routing.to.token = token.destinationSymbol[destinationNetwork] as string;
+        routing.to.network = destinationNetwork.toString().toLowerCase();
+        routing.amount = new BigNumber(amount.toString());
+
+        const transactions =
+      token.symbol.trim().toLowerCase() === "usdc"
+          ? await bridgeUSDC(
+              this.client,
+              sourceAddress,
+              destinationAdress,
+              destinationNetwork,
+              new BigNumber(amount.toString()),
+              depositAddress,
+            token as AlgorandStandardAssetConfig
```

Comment:
@AndrewC-Glitter as a third-party dev, it'd be intuitive to pass these parameters separately because the names are clear. If we use routing, we will have to explicitly mention in our docs that we use a `Routing` type and how to use it

Reply:
You can resolve this as soon as you read the above comment.

Reply:
@Elaixzar the only method that should be used by anyone for bridging transaction from Algorand is `bridgeTransactions`, which basically build transaction based on users' input, so if the input token symbol is `USDC` only `USDC` transactions are added and vice versa